### PR TITLE
feat:add playlist

### PR DESCRIPTION
**Pull request by rahulmistry751**
Added playlist funtionality
- On video listing page
- Playlist section
- Single video page

### CODE REVIEW

File: src/pages/Playlist/components/PlaylistVideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const {userToken}=useAuth();
+    const {removeVideoFromPlaylistLocal}=usePlaylistActions();
+    const removeVideoFromPlaylistHandler=(e)=>{
+        e.stopPropagation();
+        removeVideoFromPlaylistLocal(userToken,playlist._id,video._id)
+    }
+    return(
+        <div className={`card ${style['playlist-video']}`} key={video._id} onClick={()=>navigate(`/video/${video._id}`)}>
+        <div className={` ${style['card-image-container']}`}>
+            <img src={`https://img.youtube.com/vi/${video._id}/0.jpg`} alt="" />
```
Comment (shivanigangadharan) [inline src/pages/Playlist/components/PlaylistVideoCard.jsx]: can add an alt msg here

**Review by shivanigangadharan — APPROVED** [review]
Code looks good. 
Optional suggestion : after clicking checkbox while adding to playlist, you can show a toast that says video has been added to playlist.

File: src/components/PlaylistModal/PlaylistModal.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    const playlistHandler=(playlist)=>{
+        if(isVideoInPlaylist(playlist)){
+            removeVideoFromPlaylistLocal(userToken,playlist._id,currentVideo._id)
+        }
+        else{
+            addVideoToPlaylistLocal(userToken,playlist._id,currentVideo)
+        }
+    }
+    return(
```
Comment (narharikale) [inline src/components/PlaylistModal/PlaylistModal.jsx:41]: i think you need to configure your code formatter

**Review by narharikale — APPROVED** [review]
code is well structured
 can improve code readability
 try to use better code formatter

**Comment by srinibas39** [comment]
Good code structure,
Just one suggestion:you can add delete button to delete indvidual video from a playlist

Capture1

**Comment by rahulmistry751** [comment]
> Good code structure, Just one suggestion:you can add delete button to delete indvidual video from a playlist
> 
> Capture1

it's added on thumbnail

**Pull request merged**